compiler/rustc_codegen_rmc/src/codegen/intrinsic.rs

```diff
@@ -130,6 +130,61 @@ impl<'tcx> GotocCtx<'tcx> {
             }};
         }
 
+        // Intrinsics which encode a simple arithmetic operation with overflow check
+        macro_rules! codegen_op_with_overflow_check {
+            ($f:ident) => {{
+                let a = fargs.remove(0);
+                let b = fargs.remove(0);
+                let res = a.$f(b);
+                let check = Stmt::assert(
+                    res.overflowed.not(),
+                    format!("attempt to compute {} which would overflow", intrinsic).as_str(),
+                    loc,
+                );
+                let expr_place = self.codegen_expr_to_place(p, res.result);
+                Stmt::block(vec![expr_place, check], loc)
+            }};
+        }
+
+        // Intrinsics which encode a SIMD arithmetic operation with overflow check
+        // RMC overflow operations accept SIMD vectors but the results are unsound.
+        // TODO: Create a CBMC issue.
```

Comment:
Please create the issue and add it here.

Reply:
Oops. Yeah, I need to do that.

Reply:
See my comment above. I did some digging into CBMC and it doesn't look like overflow takes array as argument. I hadn't noticed that CBMC was printing a warning. I wish it would just error out instead.

Anyway, I will update the comment to reflect that.